### PR TITLE
revise the problem of makefile in the ch12

### PR DESCRIPTION
**Pull request by mudongliang**
I have seen the makefile written by pezy ! It had a little problem , I modified it !
screenshot from 2015-01-03 12 32 26

### CODE REVIEW

File: ch12/makefile [hunk 1]
```diff
@@ -2,14 +2,18 @@
 OBJECTS = ex12_06 ex12_07 ex12_10 ex12_11 ex12_13 ex12_14 ex12_15 \
 ex12_16 ex12_23 ex12_24 ex12_26 ex12_28
 
+OTHOBJS = ex12_20 ex12_27_30_TEST
+
+all:$(OBJECTS) $(OTHOBJS)
+
 # tells make to use the file "../GNU_makefile_template", which
```
Comment (pezy) [inline ch12/makefile]: `all` is already define in the template. why do I need to repeat ?

File: ch12/makefile [hunk 2]
```diff
@@ -2,14 +2,18 @@
...
 
 ex12_20: ex12_20.o ex12_19.o
 	$(CC) $(CCFLAGS) $(LOCFLAGS) ex12_20.o ex12_19.o \
-	-o ex12_20
+	-o $@
 
 ex12_27_30_TEST: ex12_27_30_TEST.o ex12_27_30.o
 	$(CC) $(CCFLAGS) $(LOCFLAGS) ex12_27_30_TEST.o ex12_27_30.o \
-	-o ex12_27_30_TEST
+	-o $@
```
Comment (pezy) [inline ch12/makefile]: any difference ?

**Comment by mudongliang** [comment]
do you make for the makefile ???

**Comment by mudongliang** [comment]
ex12_20 ex12_27_30_TEST both don't compile ! I already test it on my computer!

**Comment by mudongliang** [comment]
I'm sorry for ignoring the all in GNU_makefile_template!

**Comment by mudongliang** [comment]
I will revise it by move all in makefile of ch** to the GNU_makefile_template!

**Comment by pezy** [comment]
> ex12_20 ex12_27_30_TEST both don't compile ! I already test it on my computer!

yes, I miss it. sorry for that.

**Comment by Mooophy** [comment]
Great~

**Pull request merged**